### PR TITLE
Cleanup dialect api (again) - cross type checking

### PR DESCRIPTION
**Pull request by VitGottwald**

### CODE REVIEW

**Review by ishche — APPROVED** [review]

File: clients/cobol-dialect-api/src/index.ts [hunk 1]
```diff
@@ -1,32 +1,127 @@
 import * as vscode from "vscode";
```
Comment (ishche) [inline clients/cobol-dialect-api/src/index.ts:15]: copyright header is missing :)

Comment (VitGottwald) [reply clients/cobol-dialect-api/src/index.ts:15]: Thank you, missed that!

File: clients/idms-dialect-support/package.json [hunk 2]
```diff
@@ -73,15 +70,15 @@
     }
   },
   "scripts": {
-    "vscode:prepublish": "npm run compile",
-    "postinstall": "node -e \"if (process.env.NODE_ENV === 'production'){process.exit(1)} \" && patch-package || echo skip patching",
+    "vscode:prepublish": "npm run build",
+    "build": "esbuild src/extension.ts --bundle --sourcemap --target=node16.17.1,chrome108 --outdir=dist --external:vscode --format=cjs",
```
Comment (ishche) [inline clients/idms-dialect-support/package.json:74]: please remove "patch-package" devDependency too

**Comment by ishche** [comment]
LGTM! 👍

**Review by amanPras — APPROVED** [review]

**Pull request merged**